commit 91a9d24aa30507e6c7d8937db2de24394f0ce121
Author: Michael Wallner <mike@php.net>
Date:   Wed Apr 2 11:09:26 2014 +0200

    Fix bug #64330
    
    stream_socket_server() creates wrong Abstract Namespace UNIX sockets

--- /dev/null
+++ php5/ext/standard/tests/network/bug64330.phpt
@@ -0,0 +1,20 @@
+--TEST--
+Bug #64330 (stream_socket_server() creates wrong Abstract Namespace UNIX sockets)
+--SKIPIF--
+<?php
+if (!in_array("unix", stream_get_transports())) die("SKIP unix domain sockets unavailable");
+?>
+--FILE--
+<?php
+echo "Test\n";
+$server = stream_socket_server("unix://\x00/MyBindName");
+$client = stream_socket_client("unix://\x00/MyBindName");
+if ($client) {
+	echo "ok\n";
+}
+?>
+===DONE===
+--EXPECT--
+Test
+ok
+===DONE===
--- php5.orig/main/streams/xp_socket.c
+++ php5/main/streams/xp_socket.c
@@ -588,7 +588,8 @@ static inline int php_tcp_sockop_bind(ph
 
 		parse_unix_address(xparam, &unix_addr TSRMLS_CC);
 
-		return bind(sock->socket, (struct sockaddr *)&unix_addr, sizeof(unix_addr));
+		return bind(sock->socket, (const struct sockaddr *)&unix_addr,
+			(socklen_t) XtOffsetOf(struct sockaddr_un, sun_path) + xparam->inputs.namelen);
 	}
 #endif
 
